provider/marathon_test.go

```diff
@@ -519,57 +519,19 @@ func TestMarathonTaskFilter(t *testing.T) {
 		{
 			task: marathon.Task{
 				AppID: "specify-port-number",
-				Ports: []int{80, 443},
+				Ports: []int{8080},
 			},
 			applications: &marathon.Applications{
 				Apps: []marathon.Application{
 					{
 						ID:    "specify-port-number",
 						Ports: []int{80, 443},
 						Labels: &map[string]string{
-							"traefik.port": "80",
-						},
-					},
-				},
-			},
-			expected:         true,
-			exposedByDefault: true,
-		},
-		{
-			task: marathon.Task{
-				AppID: "specify-unknown-port-number",
-				Ports: []int{80, 443},
-			},
-			applications: &marathon.Applications{
-				Apps: []marathon.Application{
-					{
-						ID:    "specify-unknown-port-number",
-						Ports: []int{80, 443},
-						Labels: &map[string]string{
 							"traefik.port": "8080",
 						},
 					},
 				},
 			},
-			expected:         false,
-			exposedByDefault: true,
-		},
-		{
-			task: marathon.Task{
-				AppID: "specify-port-index",
-				Ports: []int{80, 443},
-			},
-			applications: &marathon.Applications{
-				Apps: []marathon.Application{
-					{
-						ID:    "specify-port-index",
```

Comment:
I believe you removed a bit too much: The index functionality is still there, hence this test case should stay.